### PR TITLE
feat(MUIv5): Upgrade SDS and MUI

### PR DESCRIPTION
**Pull request by tihuan**
- #3599 

### Reviewers
**Functional:** 
@seve 

**Readability:** 
@atarashansky 

---

## Changes
1. Update SDS to v12 and MUI to v5
2. Ran code mods to update import statements to `"@mui/material"`
3. MUIv5 has better type definitions, so updated our code accordingly
4. SDS `IconButton` is renamed to `ButtonIcon`, since the element is a button
5. A few minor styling changes to match existing prod
6. Update FMG InfoPanel to use SDS Table

    Screenshot 2022-12-13 at 6 09 21 PM

## QA steps (optional)

Please check out the rdev below, everything should work as before!

https://thuang-mui-6-frontend.rdev.single-cell.czi.technology/

~Note that WMG doesn't work in rdev, so please test it locally if needed~ it works now thanks to Alec for adding WMG data to my rdev S3! 🙏 

Thank you!

### CODE REVIEW

**Comment by atarashansky** [comment]
The rdev should work now - it was just missing data in `s3://env-rdev-wmg/thuang-mui-6/`

File: frontend/src/components/common/Filter/components/FilterContent/components/FilterRange/index.tsx [hunk 1]
```diff
@@ -47,16 +48,15 @@ export default function FilterRange({
...
-    // eslint-disable-next-line @typescript-eslint/ban-types -- {} as per Mui spec.
-    _changeEvent: ChangeEvent<{}>,
+    _changeEvent: Event,
     newRange: SliderRange
   ): void => {
     setRange(newRange);
   };
 
   const onCommittedSliderRange = (
     // eslint-disable-next-line @typescript-eslint/ban-types -- {} as per Mui spec.
```
Comment (atarashansky) [inline frontend/src/components/common/Filter/components/FilterContent/components/FilterRange/index.tsx:58]: Is this `eslint-disable` still needed?

File: frontend/src/components/common/Filter/components/FilterContent/components/FilterViews/components/FilterView/style.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ interface PanelProps {
   panelWidth: number;
 }
 
-interface ScrollProps {
+interface ScrollProps extends CommonThemeProps {
```
Comment (atarashansky) [inline frontend/src/components/common/Filter/components/FilterContent/components/FilterViews/components/FilterView/style.ts:14]: Just out of curiosity - what is the purpose of extending `CommonThemeProps`?

File: frontend/src/components/common/Filter/components/FilterSearch/index.tsx [hunk 3]
```diff
@@ -35,16 +34,12 @@ export default function FilterSearch({
       placeholder="Search"
       rightElement={
         searchValue ? (
-          <ClearIconButton
+          <ClearButtonIcon
             onClick={onClearSearch}
-            {...{
-              // TODO(cc) move this back to explicit prop={value} after upgrading SDS to enable type checking again
```
Comment (atarashansky) [inline frontend/src/components/common/Filter/components/FilterSearch/index.tsx:41]: praise be!

File: frontend/src/views/WheresMyGene/components/GeneSearchBar/components/QuickSelect/index.tsx [hunk 4]
```diff
@@ -299,15 +307,17 @@ export default function QuickSelect<
...
 
   function renderOption(
+    optionProps: HTMLAttributes<HTMLLIElement>,
     option: T,
     { selected }: AutocompleteRenderOptionState
   ) {
     return (
       <StyledMenuItem
-        {...{ component: "div" }}
+        {...{ component: "li" }}
```
Comment (atarashansky) [inline frontend/src/views/WheresMyGene/components/GeneSearchBar/components/QuickSelect/index.tsx:316]: JW, why the change from `div` to `li` here?

File: frontend/src/views/WheresMyGene/components/HeatMap/components/Chart/index.tsx [hunk 5]
```diff
@@ -304,9 +302,14 @@ export default memo(function Chart({
...
             }),
           },
-          modifiers: {
-            offset: { offset: "0,20" },
-          },
+          modifiers: [
+            {
+              name: "offset",
+              options: {
+                offset: [0, 5],
```
Comment (atarashansky) [inline frontend/src/views/WheresMyGene/components/HeatMap/components/Chart/index.tsx:309]: Is this changing the placement of the tooltip relative to the cursor?

File: frontend/src/views/WheresMyGene/components/GeneSearchBar/components/QuickSelect/index.tsx (hunk 4 above)
Comment (atarashansky) [reply frontend/src/views/WheresMyGene/components/GeneSearchBar/components/QuickSelect/index.tsx:316]: Not sure if this is related to the change from `div` to `li`, but there is this default highlight that remains on the  genes/tissues when the mouse is not hovering over an item. Also, when the selector opens up, the first item is highlighted. Is this intended or is it an unavoidable artifact? 

See video for reference:
https://user-images.githubusercontent.com/16548075/207379859-017af808-0a7e-42b2-b657-45f750fdda67.mov

**Review by atarashansky — CHANGES_REQUESTED** [review]
Amazing work, @tihuan ! This PR will make frontend work so much better going forward. 

I left a few comments, mostly just me being curious. The only thing I'd like for you to double check is the new highlighting behavior in the items of the QuickSelect component.

File: frontend/src/views/WheresMyGene/components/GeneSearchBar/components/QuickSelect/index.tsx (hunk 4 above)
Comment (tihuan) [reply frontend/src/views/WheresMyGene/components/GeneSearchBar/components/QuickSelect/index.tsx:322]: Thanks so much for catching this! I pushed a fix for the styling 🙏

And we had to use li because using div under a list (ol or ul) is technically illegal in HTML as the linter points out to me!

File: frontend/src/components/common/Filter/components/FilterContent/components/FilterViews/components/FilterView/style.ts (hunk 2 above)
Comment (tihuan) [reply frontend/src/components/common/Filter/components/FilterContent/components/FilterViews/components/FilterView/style.ts:14]: Great question! Extending CommonThemeProps is basically telling Typescript that we're using CommonThemeProps' props in our styled component, so it won't throw errors incorrectly!

File: frontend/src/views/WheresMyGene/components/HeatMap/components/Chart/index.tsx (hunk 5 above)
Comment (tihuan) [reply frontend/src/views/WheresMyGene/components/HeatMap/components/Chart/index.tsx:309]: Great eye! I had to adjust the values a little because the original values were way off in Popper2

**Review by tihuan — COMMENTED** [review]
Thanks so much for the thorough review, Alec! PTAL again 🙏

**Comment by tihuan** [comment]
> The rdev should work now - it was just missing data in `s3://env-rdev-wmg/thuang-mui-6/`

Thanks so much for copying data to S3 🙏

**Review by hthomas-czi — APPROVED** [review]
LGTM!

**Review by atarashansky — APPROVED** [review]
Fantastic work, Timmy! Looks great

**Comment by tihuan** [comment]
Thanks so much for the swift review, everyone ⭐ 🙏!
